build-system/tasks/helpers.js

```diff
@@ -453,42 +472,38 @@ async function compileUnminifiedJs(srcDir, srcFilename, destDir, options) {
     };
   }
 
-  /**
-   * Bundles an entry point with all its imports and applies babel transforms.
-   * @param {boolean} continueOnError
-   */
-  async function performBundle(continueOnError) {
-    const {banner, footer} = splitWrapper();
-    await esbuild
-      .build({
-        entryPoints: [entryPoint],
-        bundle: true,
-        sourcemap: true,
-        define: experimentDefines,
-        outfile: destFile,
-        plugins: [esbuildBabelPlugin],
-        banner,
-        footer,
-      })
-      .catch((err) => handleBundleError(err, continueOnError, destFilename));
-    finishBundle(srcFilename, destDir, destFilename, options);
-    let name = destFilename;
-    if (options.latestName) {
-      const latestMaxName = options.latestName.split('.js')[0] + '.max.js';
-      name = `${name} → ${latestMaxName}`;
-    }
-    endBuildStep('Compiled', name, startTime);
-    const target = path.basename(destFilename, path.extname(destFilename));
-    if (UNMINIFIED_TARGETS.includes(target)) {
-      await applyAmpConfig(
-        destFile,
-        /* localDev */ true,
-        /* fortesting */ options.fortesting
-      );
-    }
-  }
+  const {banner, footer} = splitWrapper();
+  const buildResult = await esbuild
+    .build({
+      entryPoints: [entryPoint],
+      bundle: true,
+      sourcemap: true,
+      define: experimentDefines,
+      outfile: destFile,
+      plugins: [esbuildBabelPlugin],
+      banner,
+      footer,
+      incremental: !!options.watch,
+    })
+    .catch((err) =>
+      handleBundleError(err, options.continueOnError, destFilename)
+    );
+  finishBundle(srcFilename, destDir, destFilename, options, startTime);
 
-  await performBundle(options.continueOnError);
+  if (options.watch) {
+    watchedTargets.set(entryPoint, {
+      rebuild: async () => {
+        const time = Date.now();
+        const buildPromise = buildResult
+          .rebuild()
+          .then(() =>
+            finishBundle(srcFilename, destDir, destFilename, options, time)
+          );
+        options?.onWatchBuild(buildPromise);
+        await buildPromise;
+      },
+    });
+  }
```

Comment:
I just noticed the watcher is not setting `continueOnError` to true. This is essential to prevent the server from bailing when it encounters its first compilation error.

Reply:
I don't think `continueOnError` is necessary anymore since all rebuilds occur out-of-band. I can fully remove it and rebuilds continue even with parse errros

Reply:
It is an essential part of minified rebuilding, since otherwise Closure will bail after the first error.

I'm a little leery of making our unminified and minified build logic diverge significantly, since it can get us back to the maintenance nightmare we dug ourselves out of a couple years ago. OTOH, if you're able to fully replace closure with `esbuild` for minified building, I'd be very happy to see all these mechanisms like the watcher and `continueOnError` get simplified, or go away to the extent possible.

WDYT about postponing this until the rewrite of minified building is complete?